docs/src/main/asciidoc/spring-cloud-gateway.adoc

```diff
@@ -509,6 +509,27 @@ spring:
 
 For a request path of `/foo/bar`, this will set the path to `/bar` before making the downstream request. Notice the `$\` which is replaced with `$` because of the YAML spec.
 
+=== SaveSession GatewayFilter Factory
+The SaveSession GatewayFilter Factory forces a `WebSession::save` operation _before_ forwarding the call downstream. This is of particular use when
+using something like http://projects.spring.io/spring-session/[Spring Session] with a lazy data store and need to ensure the session state has been saved before making the forwarded call.
+
+.application.yml
+[source,yaml]
+----
+spring:
+  cloud:
+    gateway:
+      routes:
+      - id: save_session
+        uri: http://example.org
+        predicates:
+        - Path=/foo/**
+        filters:
+        - SaveSession
+----
+
+If you are integrating http://projects.spring.io/spring-security/[Spring Security] with Spring Session, and want to ensure security details have been forwarded to the remote process, this is critical.
```

Comment:
@gregturn 
> If you are integrating Spring Security with Spring Session, ..., this is critical

In this part, does it mean that adding `SaveSession` filter is very important and it have to be applied, or adding this filter is dangerous and it should be avoided?